### PR TITLE
`TransposeLayerTest` to API2.0

### PR DESCRIPTION
**Pull request by vurusovs**

### CODE REVIEW

File: src/plugins/intel_cpu/tests/functional/shared_tests_instances/single_layer_tests/transpose.cpp [hunk 1]
```diff
@@ -4,57 +4,49 @@
 
 #include <vector>
 
-#include "single_layer_tests/transpose.hpp"
+#include "single_op_tests/transpose.hpp"
```
Comment (vurusovs) [inline src/plugins/intel_cpu/tests/functional/shared_tests_instances/single_layer_tests/transpose.cpp:7]: apply clang-format

**Review by olpipi — APPROVED** [review]

File: src/plugins/intel_cpu/tests/functional/shared_tests_instances/single_layer_tests/transpose.cpp (hunk 1 above)
Comment (ilyachur) [reply src/plugins/intel_cpu/tests/functional/shared_tests_instances/single_layer_tests/transpose.cpp:7]: Let's keep it as is. Clang-format should be enabled by plugin owner

**Review by ilyachur — APPROVED** [review]
LGTM @iefode Please take a look

File: src/tests/functional/shared_test_classes/include/shared_test_classes/single_op/transpose.hpp [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+#include <vector>
+#include <memory>
+
+#include "shared_test_classes/base/ov_subgraph.hpp"
+
+namespace ov {
+namespace test {
+typedef std::tuple<
+        std::vector<size_t>,            // Input order
+        ov::element::Type,              // Model type
```
Comment (iefode) [inline src/tests/functional/shared_test_classes/include/shared_test_classes/single_op/transpose.hpp:18]: Minor: `Model type` -> `Input type`

**Review by iefode — APPROVED** [review]

File: src/tests/functional/shared_test_classes/include/shared_test_classes/single_op/transpose.hpp (hunk 2 above)
Comment (vurusovs) [reply src/tests/functional/shared_test_classes/include/shared_test_classes/single_op/transpose.hpp:18]: Input/Output types are reserved for `inType`/`outType` test class members to be applied via `PrePostProcess`

**Pull request merged**